app/server/models/transfer_account.py

```diff
@@ -194,19 +194,38 @@ def get_approval(self, receiving_address):
                 return approval
         return None
 
-    def approve_and_disburse(self):
-        if not self.is_approved:
-            self.is_approved = True
+    def approve_and_disburse(self, initial_disbursement=None, auto_resolve=False):
+        from server.utils.access_control import AccessControl
 
-            if self.is_beneficiary:
-                disbursement = self.make_initial_disbursement()
+        auto_approve = current_app.config['AUTO_APPROVE_TRANSFER_ACCOUNTS']
+        admin = getattr(g, 'user', None)
+
+        if not self.is_approved and admin:
+            if AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'superadmin'):
+                auto_resolve = True
+                self.is_approved = True
+
+            elif AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'admin'):
+                auto_resolve = initial_disbursement \
+                               <= current_app.config['MAXIMUM_CUSTOM_INITIAL_DISBURSEMENT']
+                self.is_approved = auto_approve
+
+            elif AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'subadmin'):
+                auto_resolve = False
+                self.is_approved = False
+
+        if self.is_beneficiary:
+            if len(self.credit_receives) < 1:
+                # approve_and_disburse might be called for a second time to disburse
+                # so first check that no credit transfer have already been received
```

Comment:
when does this occur?

Reply:
ohh when admin calls it.

I think it makes more sense to seperate out these steps more, because right now:
- a superadmin can't approve a user but reject a non-default amount.
- a subadmin can call this function from the 'approve' section and it'll return success, when really it has rejected them
- if a user who's yet to be activated receives credits from another party, this will incorrectly assume they've received a disbursement,

Something like the following probably makes more sense IMO:
1. Have a function that contains most of this code that fires when a user is first created -
2. Have a second approval function that admins and superadmins can call, that approves (and probably also resolves the initial disbursement)
3. leave non default amounts to be manually approved

Reply:
> a subadmin can call this function from the 'approve' section and it'll return success, when really it has rejected them
Subadmins can't approve users... ? `@requires_auth(allowed_roles={'ADMIN': 'admin'})`

> if a user who's yet to be activated receives credits from another party, this will incorrectly assume they've received a disbursement,
That is true

> Have a second approval function that admins and superadmins can call, that approves (and probably also resolves the initial disbursement)
I don't think we need a second approval function? I agree there could be better handling in how we resolve initial disbursement

Reply:
> I don't think we need a second approval function? I agree there could be better handling in how we resolve initial disbursement

"second approval function" was bad wording on my behalf. more of "A second function in addition to the one mention above, that is specifically designed as an approval one". Definitely open to other implementions, but keen to obey Single Responsibility Prinicple

Reply:
but would `approve` resolve an initial disbursement or is that done separately? If a super admin resolves an initial disbursement as COMPLETE, but the user is unapproved, is that a big deal?

Reply:
How would you know when to fire the second function  that only approves?

Reply:
> If a super admin resolves an initial disbursement as COMPLETE, but the user is unapproved, is that a big deal?

I don't think so.

> but would approve resolve an initial disbursement or is that done separately? 

I can see this going one of three ways:

- always resolve initial disbursement (if user has sufficient authority). If you want to reject an initial disbursement you need to do it before approving a user. (This is pretty much what you've implemented above)
Pros: For superadmins, consistent behaviour regardless of whether the disbursement is default or not.
Cons: Easy to accidentally approve a bad disbursement. If an admin is approving a user with non-default amount, then the amount won't be resolved, so inconsistent behaviour there.

- never resolve an initial disbursement
Pros: Consistent behaviour with no surprises. Won't ever result in superadmin approving a non default amount when they didn't mean to.
Cons: Potentially doubles the required steps required to activate a user

- only resolve initial disbursement when it's default
Pros: Admins and superadmins both experience the same behaviour. Good balance between ease of activating users, and control over disbursement amounts.
Cons: Kinda hard to predict behaviour from point of view of approver if u don't remember whether amount is default or not.

Reply:
No strong opinion over which one is best. In the  absence of any datapoint, it'd probably just choose the path that's easiest

Reply:
> only resolve initial disbursement when it's default
I think this makes the most sense